backend/src/common/services/auth.ts

```diff
@@ -0,0 +1,35 @@
+import axios from 'axios'
+import SlackClientService from "./slackClient"
+
+interface IAuthTestResponse {
+  ok: boolean
+  url: string
+  team: string
+  user: string
+  team_id: string
+  user_id: string
+}
+
+export default class AuthService {
+  private slackClientService: SlackClientService
+
+  constructor() {
+    this.slackClientService = new SlackClientService()
+  }
+
+  public async checkAuth(token: string) {
+    return await this.slackClientService
+      .checkAuth(token) as IAuthTestResponse
+  }
+
+  public async login(code: string) {
+    return await axios.get('https://slack.com/api/oauth.access', {
```

Comment:
Url to consts?

Reply:
axios methods are generics, you can use them like this
`axios.get<ReturnType>()`

also return `data` field from this method to avoid destructuring later